### PR TITLE
Multiple transaction versions

### PR DESCRIPTION
**Pull request by raynaudoe**
- Adds multiple transaction versions
- Update testcases.
- Fix types `read` and `toString` code

### CODE REVIEW

File: app/src/coin.h [hunk 1]
```diff
@@ -42,8 +42,9 @@ typedef enum {
 
 // Coin Specific
 #define PK_ADDRESS_TYPE                     COIN_ADDR_TYPE_KUSAMA
-#define SUPPORTED_TX_VERSION                LEDGER_MAJOR_VERSION
-#define SUPPORTED_SPEC_VERSION              (2000+LEDGER_MINOR_VERSION)
+#define SUPPORTED_TX_VERSION_CURRENT        LEDGER_MAJOR_VERSION
+#define SUPPORTED_TX_VERSION_PREVIOUS       (LEDGER_MAJOR_VERSION - 1)
+#define SUPPORTED_SPEC_VERSION              LEDGER_MINOR_VERSION
```
Comment (jleni) [inline app/src/coin.h:47]: We may have a problem here.

Unfortunately, ledger does not support version numbers > 255. But Kusama uses Major version numbers >2000, for this reason the value in the makefile needs to be increased by 2000. Removing that will probably result in incorrect checks.

File: app/src/parser_impl.c [hunk 2]
```diff
@@ -345,14 +344,18 @@ parser_error_t _checkVersions(parser_context_t *c) {
...
 
-    if (specVersion < SUPPORTED_MINIMUM_SPEC_VERSION) {
-        return parser_spec_not_supported;
+    if (transactionVersion != (SUPPORTED_TX_VERSION_CURRENT) &&
+        transactionVersion != (SUPPORTED_TX_VERSION_PREVIOUS) ) {
+        return parser_tx_version_not_supported;
     }
 
-    if (transactionVersion != (SUPPORTED_TX_VERSION)) {
+    if (specVersion < SUPPORTED_MINIMUM_SPEC_VERSION) {
```
Comment (jleni) [inline app/src/parser_impl.c:352]: here we may have trouble as described in the previous comment

**Review by jleni — CHANGES_REQUESTED** [review]
beautiful improvement! great work!

File: app/src/coin.h (hunk 1 above)
Comment (raynaudoe) [reply app/src/coin.h]: ok, fixed it

**Pull request merged**